functions.php

```diff
@@ -53,4 +54,65 @@ function displayGames(array $games): string
     return $result;
 }
 
+/**
+ * validate form inputs
+ *
+ * @param string $name
+ * @param string $year
+ * @param string $developer
+ * @param string $imdb_rating
+ * @param string $image_url
+ * @return string 'Valid' if all the inputs are valid and 'Invalid' if more than one field is invalid
+ */
+function validateFormInput(string $name, string $year, string $developer, string $imdb_rating, string $image_url): string
+{
+    $validName = filter_var($name, FILTER_SANITIZE_STRING);
+    $validName = preg_match('/^[a-zA-Z0-9 ]{1,255}$/', $name);
+
+    $validYear = filter_var($year, FILTER_SANITIZE_NUMBER_INT);
+    $validYear = filter_var($year, FILTER_VALIDATE_INT, ["options" => ["min_range" => 1000, "max_range" => 2023]]);
+
+    $validDeveloper = filter_var($developer, FILTER_SANITIZE_STRING);
+    $validDeveloper = preg_match('/^[a-zA-Z ]{1,255}$/', $developer);
+
+    $validImdb_rating = filter_var($imdb_rating, FILTER_SANITIZE_NUMBER_FLOAT);
+    $validImdb_rating = filter_var($imdb_rating, FILTER_VALIDATE_FLOAT, ["options" => ["min_range" => 1.0, "max_range" => 10.0]]);
+
+    $validImage_url = filter_var($image_url, FILTER_SANITIZE_URL);
+    $validImage_url = filter_var($image_url, FILTER_VALIDATE_URL);
+
+    if ($validName && $validYear && $validDeveloper && $validImdb_rating && $validImage_url) {
+        return 'Valid';
+    } elseif($validName && $validYear && $validDeveloper && $validImdb_rating &&! $validImage_url) {
+        return 'Invalid URL';
+    } elseif($validName && $validYear && $validDeveloper && $validImage_url &&! $validImdb_rating) {
+        return 'Invalid rating'; // turn to float??
+    } elseif($validName && $validYear && $validImdb_rating && $validImage_url &&! $validDeveloper) {
+        return 'Invalid Developer name. A-Z only';
+    } elseif($validName && $validDeveloper && $validImdb_rating && $validImage_url &&! $validYear) {
+        return 'Invalid year';
+    } elseif($validYear && $validDeveloper && $validImdb_rating && $validImage_url &&! $validName) {
+        return 'Invalid Game name. Please remove special characters';
+    } else {
+        return 'Invalid';
+    }
+}
+
+/**
```

Comment:
strictly speaking this shouldn't be in this branch

Reply:
Forgot about that - will remove